src/heap.py

```diff
@@ -43,15 +45,17 @@ def pop(self):
         try:
             return_val = self.heap[0]
             del self.heap[0]
-            self.heap.insert(0, self.heap.pop(len(self.heap) - 1))
-            # self._sift_down()
+            self.heap.insert(0, self.heap.pop(self.length() - 1))
```

Comment:
Consider adding a `__len__` special method to your class. https://docs.python.org/3/reference/datamodel.html#object.__len__